concrete/src/Page/Page.php

```diff
@@ -1399,7 +1407,7 @@ public function getCollectionLink($appendBaseURL = false)
      */
     public function getSiteTreeID()
     {
-        return $this->siteTreeID;
+        return $this->cPointerOriginalSiteTreeID ?: $this->siteTreeID;
```

Comment:
How can we get the `siteTreeID` of the alias?

Reply:
`populatePage()` does that, the same way it gets the original pointer cID

Reply:
Wouldn't that affect `\Localization::activeLocale()`? With the change I propose this is also fixed. e.g. an alias inside another site tree will then also show the correct locale for the site tree it is in. If you change this for the icon only you'd probably want to change this to retrieve the locale as well.

This was also the 5.7.5.13 behavior.

Reply:
Let's let @aembler decide about that :wink:

Reply:
I do understand you'd want the `getSiteTreeID()` to give a predictable result if the cID is a certain value. But `getByID()` doesn't return the requested page either if it is an alias, so `getSiteTreeID()` is inconsistent with what you'd expect either way.

Suppose cID 101 is an alias of 100;
```php
$original = \Page::getByID(100);
$alias = \Page::getByID(101);

echo $original->getCollectionID(); // outputs 100
echo $alias->getCollectionID(); // outputs 100
```

I'm not sure what broke the behavior, but I suspect these problems where already there then as well. To properly fix all of this seems like a large task?

Reply:
> I suspect these kind of inconsistencies where already there in 5.7 as well

Yes, it has always been like that. It's not an inconsistency, it's how page aliases works: we load the destination page, not the alias: [this is the only data](https://github.com/concrete5/concrete5/blob/83bfd84020c45d2383b8fdb730cceec9416a1693/concrete/src/Page/Page.php#L262-L267) we read from the alias itself.

> 'm not sure what broke retreiving the correct locale for aliasses

Maybe the `TreeInterface` interface?